src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncReceiptsRequestHandler.scala

```diff
@@ -58,6 +76,30 @@ class FastSyncReceiptsRequestHandler(
     }
   }
 
+  /**
+    * Validates whether the received receipts match the block headers stored on the blockchain,
+    * returning the valid receipts
+    *
+    * @param requestedHashes hash of the blocks to which the requested receipts should belong
+    * @param receipts received by the peer
+    * @return the valid receipts or the error encountered while validating them
+    */
+  private def validateReceipts(requestedHashes: Seq[ByteString], receipts: Receipts): ReceiptsValidationResult = {
+    val blockHashesWithReceipts = requestedHashes.zip(receipts.receiptsForBlocks)
+    val blockHeadersWithReceipts = blockHashesWithReceipts.map{ case (hash, blockReceipts) =>
+      blockchain.getBlockHeaderByHash(hash) -> blockReceipts }
```

Comment:
Due to a recent [PR](https://github.com/input-output-hk/mantis/pull/280) that changed in how we enqueue the request to download, we currently only request receipts if the header was saved into the db https://github.com/input-output-hk/mantis/blob/phase/daedalus/src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala#L350).